### DISCUSSION TITLE
Python (NO TTX): Opening and then saving a TTF generates differing files

### DISCUSSION THREAD

**Discussion by UnBeatWaterGH** [fonttools/fonttools#3444] [Q&A]
Hashes are different, hashes should be the same

Original: `e232c3ad70b9f5cf386c51f316caba3542332cddb5bb58cae96de3eaf1d4e1af47a17c769ee5e30c371102cac537fc1d5abbe3577befc61c7c8df5d211b1dd32`
Open then save: `6b9a44a24b8afc3c573dd382bb63809083fbc08a4969359e929ba1cd6249c90394e356be6d144cd4d5a4ea84b2fcc541cb2d241886ac5283867d0c90a1a9a356`

Fyi, the reason why I'm using fonttools is to change the "family" of a TTF file, that's the only reason I'm trying to use fonttools for, I hate TTX, because it creates an intermediary file.

**Comment by kenmcd** [comment]
Probably the recalculation of the timestamp.

See similar issue here: https://github.com/fonttools/fonttools/issues/3385

See the Compile Options here: 
https://fonttools.readthedocs.io/en/latest/ttx.html#compile-options

**Comment by UnBeatWaterGH** [comment]
> Probably the recalculation of the timestamp.
> 
> See similar issue here: #3385
> 
> See the Compile Options here: https://fonttools.readthedocs.io/en/latest/ttx.html#compile-options

not using ttx, tried ttx, it had the same problem as python api
I HATE TTX WHY DOES IT HAVE TO GENERATE AN XML, WHY CANT IT JUST CHANGE THE FONT FAMILY DIRECTLY

**Comment by UnBeatWaterGH** [comment]
im using fonttools module

**Comment by UnBeatWaterGH** [comment]
not ttx module

**Comment by kenmcd** [comment]
You could extract both fonts using `ttx`, and then do a compare to look for the differences.
That would give a clue to the source of the difference.

`gftools` also has a `diff` tool.

One of the foottools guys can probably give you a more definite answer.
So wait for them to reply...

**Reply by UnBeatWaterGH** [reply]
Foottools, we provide tools, for your feet.

**Comment by UnBeatWaterGH** [comment]
sure, but im not using ttx
i hate ttx

**Comment by UnBeatWaterGH** [comment]
```
--- Mxtx.ttx	2024-02-09 01:14:37.167389181 +0100
+++ Mxtx2.ttx	2024-02-09 01:15:39.747828321 +0100
@@ -304,12 +304,12 @@
     <!-- Most of this table will be recalculated by the compiler -->
     <tableVersion value="1.0"/>
     <fontRevision value="1.0"/>
-    <checkSumAdjustment value="0x87386977"/>
+    <checkSumAdjustment value="0xc6c89800"/>
     <magicNumber value="0x5f0f3cf5"/>
     <flags value="00000000 00001011"/>
     <unitsPerEm value="1600"/>
     <created value="Wed Sep 30 22:09:33 2015"/>
-    <modified value="Mon May 18 15:14:29 2020"/>
+    <modified value="Fri Feb  9 00:14:37 2024"/>
     <xMin value="0"/>
     <yMin value="-400"/>
     <xMax value="800"/>
@@ -338,7 +338,7 @@
     <reserved2 value="0"/>
     <reserved3 value="0"/>
     <metricDataFormat value="0"/>
-    <numberOfHMetrics value="4"/>
+    <numberOfHMetrics value="1"/>
   </hhea>
 
   <maxp>
```

**Comment by UnBeatWaterGH** [comment]
@kenmcd

**Comment by UnBeatWaterGH** [comment]
[Mxtx.ttx](https://github.com/fonttools/fonttools/files/14215883/Mxtx.txt)
[Mxtx2.ttx](https://github.com/fonttools/fonttools/files/14215884/Mxtx2.txt)
